integration/loki_micro_services_delete_test.go

```diff
@@ -247,13 +247,7 @@ func TestMicroServicesDeleteRequest(t *testing.T) {
 		require.Eventually(t, func() bool {
 			deleteRequests, err := cliCompactor.GetDeleteRequests()
 			require.NoError(t, err)
-			require.Len(t, deleteRequests, len(expectedDeleteRequests))
-			for i := range deleteRequests {
-				if deleteRequests[i] != expectedDeleteRequests[i] {
-					return false
-				}
-			}
-			return true
+			return assert.ElementsMatch(t, client.DeleteRequests(expectedDeleteRequests), deleteRequests)
```

Comment:
`assert.ElementsMatch()` causes the test to fail, even though it returns a `bool`. That's very bad API desigh IMO.